test/RFClassifyTest.ecl

```diff
@@ -0,0 +1,127 @@
+  // Pick random X2 and X3: Choose val between 0 and 1 and round to 0 or 1.
+  SELF.X2 := ROUND(r2%1000000 / 1000000);
+  BOOLEAN x2B := SELF.X2=1;
+  SELF.X3 := ROUND(r3%1000000 / 1000000);
+  BOOLEAN x3B := SELF.X3=1;
+  BOOLEAN y := MAP(SELF.X1 < -50 => x2B OR x3B, // OR
+                         SELF.X1 >= -50 AND SELF.X1 < 0 => x2B AND x3B, // AND
+                         SELF.X1 >= 0 AND SELF.X1 < 50 => (x2B OR x3B) AND (NOT (x2B AND x3B)), // XOR
+                         (NOT (x2B OR x3B)));  // NOR
+  SELF.Y := IF(y, 1, 0);
+END;
+ds := NORMALIZE(dummy, numTrainingRecs, make_data(LEFT, COUNTER));
+OUTPUT(ds, NAMED('TrainingData'));
+
+X1 := PROJECT(ds, TRANSFORM(NumericField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 1,
+                            SELF.value := LEFT.X1));
+X2 := PROJECT(ds, TRANSFORM(NumericField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 2,
+                            SELF.value := LEFT.X2));
+X3 := PROJECT(ds, TRANSFORM(NumericField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 3,
+                            SELF.value := LEFT.X3));
+// Add noise to Y by randomly flipping the value according to PROBABILITY(errorProb).
+Y := PROJECT(ds, TRANSFORM(DiscreteField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 1,
+                            SELF.value := IF(prob(errorProb), (LEFT.Y + 1)%2, LEFT.Y)));
+nominals := [];
+X := X1 + X2 + X3;
+
+// Expand to number of work items
+Xe := NORMALIZE(X, wiCount, TRANSFORM(NumericField, SELF.wi := COUNTER, SELF := LEFT));
+Ye := NORMALIZE(Y, wiCount, TRANSFORM(DiscreteField, SELF.wi := COUNTER, SELF := LEFT));
+
+OUTPUT(Ye, NAMED('Y_train'));
+
+F := LT.ClassificationForest(numTrees, numVarsPerTree);
+
+//mod := F.GetModel(Xe, Ye, nominals);
+
+// With this line it runs fine.
+mod := F.GetModel(X, Y, nominals);
+
+OUTPUT(mod, NAMED('Model'));
+modStats := F.GetModelStats(mod);
+OUTPUT(modStats, NAMED('ModelStats'));
+tn0 := F.Model2Nodes(mod);
+tn := SORT(tn0, wi, treeId, level, nodeid, id, number, depend);
+OUTPUT(tn, {wi, level, treeId, nodeId, parentId, isLeft, id, number, value, depend, support, isOrdinal},NAMED('Tree'));
+
+dsTest := DISTRIBUTE(SORT(NORMALIZE(dummy, numTestRecs, make_data(LEFT, COUNTER)), id, LOCAL), id);
+X1t := PROJECT(dsTest, TRANSFORM(NumericField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 1,
+                            SELF.value := LEFT.X1));
+X2t := PROJECT(dsTest, TRANSFORM(NumericField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 2,
+                            SELF.value := LEFT.X2));
+X3t := PROJECT(dsTest, TRANSFORM(NumericField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 3,
+                            SELF.value := LEFT.X3));
+Xt := X1t + X2t + X3t;
+Ycmp := PROJECT(dsTest, TRANSFORM(DiscreteField, SELF.wi := 1, SELF.id := LEFT.id, SELF.number := 1,
+                            SELF.value := LEFT.Y));
+Yhat0 := F.Classify(Xt, mod);
+Yhat := DISTRIBUTE(SORT(Yhat0, id, LOCAL),  id);
```

Comment:
A distribute after a sort seems weird.  What are you trying to accomplish?  I would have expected to see SRT(DISTRIBUTE(... instead.